xen/arch/arm/domctl.c

```diff
@@ -48,6 +49,16 @@ static int handle_vuart_init(struct domain *d,
     return rc;
 }
 
+static int get_sci_info(struct domain *d, struct xen_domctl_sci_info *sci_info)
+{
+#ifdef CONFIG_ARM_SCI
+    sci_info->paddr = d->arch.sci_channel.paddr;
+    sci_info->func_id = d->arch.sci_channel.guest_func_id;
+#endif /* CONFIG_ARM_SCI */
+    return 0;
```

Comment:
I believe you need to return `-ENODEV` if `CONFIG_ARM_SCI` is not enabled or SCI is not enabled for the domain

Reply:
fixed

Reply:
What about the second part of my comment? What if SCI is not enabled for that particular domain? It will have no valid paddr and func id in that case